packages/lib/src/vis/utils.ts

```diff
@@ -348,3 +347,30 @@ export function checkModifierKey(
 
   return event.getModifierState(modifierKey);
 }
+
+export function getRatioEndPoint(
+  startPoint: Vector2,
+  endPoint: Vector2,
+  ratio: number
+) {
+  const widthSign = Math.sign(endPoint.x - startPoint.x);
+  const width = Math.abs(endPoint.x - startPoint.x);
+
+  const heightSign = Math.sign(endPoint.y - startPoint.y);
+  const height = Math.abs(endPoint.y - startPoint.y);
+
+  const originalRatio =
+    Math.abs(endPoint.x - startPoint.x) / Math.abs(endPoint.y - startPoint.y);
+
+  if (originalRatio < ratio) {
+    return new Vector2(
+      startPoint.x + widthSign * height * ratio,
+      startPoint.y + heightSign * height
+    );
+  }
+
+  return new Vector2(
+    startPoint.x + widthSign * width,
+    startPoint.y + (heightSign * width) / ratio
+  );
+}
```

Comment:
Contrary to `getSizeToFit` that returns a size included in the input size, this utility changes the `endPoint` so that the resulting rectangle **includes** the original rectangle.